desktop/core/src/desktop/js/apps/notebook/app.js

```diff
@@ -953,7 +954,21 @@ huePubSub.subscribe('app.dom.loaded', app => {
         huePubSub.subscribe(
           'editor.upload.query',
           query_id => {
-            viewModel.selectedNotebook().snippets()[0].uploadQuery(query_id);
+            const useInternalHueOptimizerStorage = true;
+            if (useInternalHueOptimizerStorage) {
+              const snippet = viewModel.selectedNotebook().snippets()[0];
+              const dialect = snippet.dialect();
+              const statement = snippet.statement_raw();
+              sqlParserRepository.getAutocompleteParser(dialect).then(parser => {
+                const parseResults = parser.parseSql(statement, '');
+                $.post('/metadata/api/optimizer/upload/query', {
+                  sourcePlatform: dialect,
+                  locations: JSON.stringify(parseResults.locations)
+                });
+              });
```

Comment:
I would refactor this all new code by placing it in a well named function that you call from here. Preferably in its own file so that we don't keep keep adding logic to this huge js file.